File: Exercises/101-109 - Small Problems/easy_4/running_totals.rb
# Write a method that takes an Array of numbers, and returns an Array with the same number of elements,
# and each element has the running total from the original Arra

def running_total(numbers)
  numbers.each.with_index do |num, i|
    if i != 0
      numbers[i] += numbers[i - 1]
    end
  end
  numbers
end

puts running_total([2, 5, 13]) == [2, 7, 20]
puts running_total([14, 11, 7, 15, 20]) == [14, 25, 32, 47, 67]
puts running_total([3]) == [3]
puts running_total([]) == []

File: Exercises/210 - Small Problems/easy-3/running-totals.js
/*
Write a function that takes an array of numbers, and returns an array with the same number of elements with each elements value being running total from the original array.

Examples:

runningTotal([2, 5, 13]);             // [2, 7, 20]
runningTotal([14, 11, 7, 15, 20]);    // [14, 25, 32, 47, 67]
runningTotal([3]);                    // [3]
runningTotal([]);                     // []
*/

function runningTotal(nums) {
  return nums.map(function(value, index) {
    return nums.slice(0, index + 1).reduce(function(total, values) {
      return total + values;
    });
  });
}

console.log(runningTotal([2, 5, 13]));
console.log(runningTotal([14, 11, 7, 15, 20]));
console.log(runningTotal([3]));
console.log(runningTotal([]));